rmgpy/molecule/resonance.py

```diff
@@ -341,17 +389,187 @@ def generate_isomorphic_isomers(mol):
-    )
+    cython.declare(output=list, molList=list, newmol=Molecule, SSSR=list, bonds=list, solution=list,
+                   y=list, x=list, index=cython.int, bond=Bond, ring=list)
+
+    if not mol.isCyclic():
+        return []
+
+    output = clarOptimization(mol)
+
+    molList = []
+
+    for newmol, SSSR, bonds, solution in output:
+
+        # The solution includes a part corresponding to rings, y, and a part corresponding to bonds, x, using
+        # nomenclature from the paper. In y, 1 means the ring as a sextet, 0 means it does not.
+        # In x, 1 corresponds to a double bond, 0 either means a single bond or the bond is part of a sextet.
+        y = solution[0:len(SSSR)]
+        x = solution[len(SSSR):]
+
+        # Apply results to molecule - double bond locations first
+        for index, bond in enumerate(bonds):
+            if x[index] == 0:
+                bond.order = 'S'
+            elif x[index] == 1:
+                bond.order = 'D'
+            else:
+                raise ValueError('Unaccepted bond value {0} obtained from optimization.'.format(x[index]))
+
+        # Then apply locations of aromatic sextets by converting to benzene bonds
+        for index, ring in enumerate(SSSR):
+            if y[index] == 1:
+                clarTransformation(newmol, ring)
+
+        try:
+            newmol.updateAtomTypes()
+        except AtomTypeError:
+            pass
+        else:
+            molList.append(newmol)
+
+    return molList
+
+
+def clarOptimization(mol, constraints=None, maxNum=None):
+    """
+    Implements linear programming algorithm for finding Clar structures. This algorithm maximizes the number
+    of Clar sextets within the constraints of molecular geometry and atom valency.
+
+    Method from:
+        Hansen, P.; Zheng, M. The Clar Number of a Benzenoid Hydrocarbon and Linear Programming.
+            J. Math. Chem. 1994, 15 (1), 93–107.
+    """
+    cython.declare(molecule=Molecule, ASSSR=list, exo=list, l=cython.int, m=cython.int, n=cython.int,
+                   a=list, objective=list, status=cython.int, solution=list, innerSolutions=list)
+
+    from lpsolve55 import lpsolve
+
+    # Make a copy of the molecule so we don't destroy the original
+    molecule = mol.copy(deep=True)
+
+    ASSSR = molecule.getAromaticSSSR()
+
+    if not ASSSR:
+        return []
+
+    # Get list of atoms that are in rings
+    atoms = set()
+    for ring in ASSSR:
+        atoms.update(ring)
+    atoms = list(atoms)
+
+    # Get list of bonds involving the ring atoms, ignoring bonds to hydrogen
+    bonds = set()
+    for atom in atoms:
+        bonds.update([atom.bonds[key] for key in atom.bonds.keys() if key.isNonHydrogen()])
+    bonds = list(bonds)
+
+    # Identify exocyclic bonds, and save their bond orders
+    exo = []
+    for bond in bonds:
+        if bond.atom1 not in atoms or bond.atom2 not in atoms:
+            if bond.order == 'D':
+                exo.append(1)
```

Comment:
At this point, exocyclic bonds should only be either single or double bonds. If it were a benzene bond then the bond should be included in the list of aromatic rings.